test/table.js

```diff
@@ -1379,6 +1379,253 @@ describe('Bigtable/Table', function() {
     });
   });
 
+  describe('waitForReplication', function() {
+    it('should return the error to the callback', function(done) {
+      var error = new Error('err');
+
+      table.bigtable.request = function(config, callback) {
+        callback(error);
+      };
+
+      table.waitForReplication(function(err) {
+        assert.strictEqual(err, error);
+        done();
+      });
+    });
+
+    it('should call generateConsistencyToken', done => {
+      table.generateConsistencyToken = callback => {
+        assert.strictEqual('function', typeof callback);
+        done();
+      };
+      table.waitForReplication();
+    });
+
+    it('should call checkConsistency', done => {
+      var response = {
+        consistency_token: 'sample-token12345',
+      };
+
+      table.bigtable.request = function(config, callback) {
+        callback(null, response);
+      };
+
+      table.checkConsistency = function(token, callback) {
+        assert.strictEqual(token, response.consistency_token);
+        callback(null, {consistent: true});
+      };
+
+      table.waitForReplication(() => {
+        done();
+      });
+    });
+
+    describe('retiries', () => {
+      let clock;
+      let setTimeoutSpy;
+      let clearTimeoutSpy;
+      let checkConsistencySpy;
+
+      beforeEach(() => {
+        clock = sinon.useFakeTimers({
+          toFake: ['setTimeout', 'clearTimeout'],
+        });
+        setTimeoutSpy = sinon.spy(global, 'setTimeout');
+        clearTimeoutSpy = sinon.spy(global, 'clearTimeout');
+        checkConsistencySpy = sinon.spy(table, 'checkConsistency');
+      });
+
+      afterEach(() => {
+        clock.restore();
+      });
+
+      it('should return true if token is consistent', function(done) {
+        table.bigtable.request = function(config, callback) {
+          if (config.method === 'generateConsistencyToken') {
+            // firstCall = false;
+            return callback(null, [{consistency_token: 'sample-token12345'}]);
+          }
+          if (config.method === 'checkConsistency') {
+            return callback(null, {consistent: true});
+          }
+        };
+
+        table.waitForReplication(function(err, response) {
+          // Checks that a 10 minute timer was set.
+          setTimeoutSpy.calledWith(sinon.match.func, 10 * 60 * 1000);
+
+          // check checkConsistencySpy called for first time
+          checkConsistencySpy.called;
+
+          // Checks that clearInterval was called.
+          clearTimeoutSpy.called; // true;
+          assert.strictEqual(response, true);
+          assert.ifError(err);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should retry checkConsistency', done => {
+        this.timeout(6000);
```

Comment:
This mocha directive shouldn't be needed since the tests isn't actually taking this long